manager/src/main/java/com/redhat/developer/manager/ProcessorService.java

```diff
@@ -0,0 +1,53 @@
+package com.redhat.developer.manager;
+
+import java.time.ZonedDateTime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import com.redhat.developer.infra.dto.BridgeStatus;
+import com.redhat.developer.manager.api.models.requests.ProcessorRequest;
+import com.redhat.developer.manager.dao.BridgeDAO;
+import com.redhat.developer.manager.dao.ProcessorDAO;
+import com.redhat.developer.manager.exceptions.AlreadyExistingItemException;
+import com.redhat.developer.manager.exceptions.BridgeLifecycleException;
+import com.redhat.developer.manager.exceptions.ItemNotFoundException;
+import com.redhat.developer.manager.models.Bridge;
+import com.redhat.developer.manager.models.Processor;
+
+@ApplicationScoped
+public class ProcessorService {
+
+    @Inject
+    ProcessorDAO processorDAO;
+
+    @Inject
+    BridgeDAO bridgeDAO;
+
+    @Transactional
+    public Processor createProcessor(String customerId, String bridgeId, ProcessorRequest processorRequest) {
+        Bridge bridge = bridgeDAO.findByIdAndCustomerId(bridgeId, customerId);
+        if (bridge == null) {
+            throw new ItemNotFoundException("Bridge with id '" + bridgeId + "' does not exist for customer '" + customerId + "'");
+        }
+
+        if (BridgeStatus.AVAILABLE != bridge.getStatus()) {
+            /* We cannot deploy Processors to a Bridge that is not Available */
+            throw new BridgeLifecycleException("Bridge with id '" + bridge.getId() + "' for customer '" + customerId + "' is not in the '" + BridgeStatus.AVAILABLE + "' state.");
+        }
+
+        Processor p = processorDAO.findByBridgeIdAndName(bridgeId, processorRequest.getName());
+        if (p != null) {
+            throw new AlreadyExistingItemException("Processor with name '" + processorRequest.getName() + "' already exists for bridge with id '" + bridgeId + "' for customer '" + customerId + "'");
+        }
+
+        p = new Processor();
+        p.setName(processorRequest.getName());
+        p.setSubmittedAt(ZonedDateTime.now());
+        p.setStatus(BridgeStatus.REQUESTED);
+        bridge.addProcessor(p);
```

Comment:
```suggestion
```
According to our schema definition, I think you don't need to do this

Reply:
This is setting up the foreign key relationship between Bridge --> Processor as part of the data i.e. a Processor cannot be inserted into the DB without being associated with a valid Bridge.

Reply:
You sure @robpblake ? I've just tried and it works

Reply:
You make me doubt myself now. Let me double check.

Reply:
@r00ta Just to double check: you're suggesting I remove the `bridge.addProcessor(p);` line?

When I do that I get (as I would expect):

```
Caused by: org.h2.jdbc.JdbcSQLException: NULL not allowed for column "BRIDGE_ID"; SQL statement:
insert into Processor (bridge_id, name, published_at, status, submitted_at, version, id) values (?, ?, ?, ?, ?, ?, ?) [23502-197]
```

Nothing else is setting up the relationship between the entities other that the `addProcessor(p)` call.

Reply:
Let me try again

Reply:
Sorry @robpblake , I mean to replace `bridge.addProcessor(p);` with `p.setBridge(bridge);`. In this very moment `bridge.addProcessor(p);` is setting adding the processor `p` to its list and the bridge `bridge` to the processor `p`. We can remove this if you simply do `p.setBridge(bridge);` here